app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -397,10 +400,13 @@ private void showMenu() {
     }
 
     private void updateMenu() {
+
         final boolean isMyShotUnreadEnabled = AppConfigWrapper.getMyshotUnreadEnabled(this);
         final boolean showUnread = isMyShotUnreadEnabled && Settings.getInstance(this).hasUnreadMyShot();
         myshotIndicator.setVisibility(showUnread ? View.VISIBLE : View.GONE);
         if (pendingMyShotOnBoarding) {
+            pendingMyShotOnBoarding = false;
+            setShowNightModeSpotlight(false);
```

Comment:
I feel we have a need to write tests to protect the logic here. I've created #2785 for this.